projects/canopy/src/lib/forms/sort-code/sort-code.component.spec.ts

```diff
@@ -0,0 +1,305 @@
+    });
+
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+
+    it('links the hint to the fieldset with the correct aria attributes', () => {
+      expect(fieldsetElement.nativeElement.getAttribute('aria-describedby')).toContain(
+        hintId,
+      );
+    });
+
+    it('links custom error messages to the input field with the correct aria attributes', () => {
+      expect(fieldsetElement.nativeElement.getAttribute('aria-describedby')).toContain(
+        errorId,
+      );
+    });
+
+    it('combines both the hint and error ids to create the aria described attribute', () => {
+      expect(fieldsetElement.nativeElement.getAttribute('aria-describedby')).toBe(
+        `${hintId} ${errorId}`,
+      );
+    });
+  });
+
+  it('sets the individual input fields when a sort code value is provided', () => {
+    component.form.get('sortCode').setValue('204060');
+    fixture.detectChanges();
+    expect(firstInput.nativeElement.value).toEqual('20');
+    expect(secondInput.nativeElement.value).toEqual('40');
+    expect(thirdInput.nativeElement.value).toEqual('60');
+  });
+
+  it('joins the individual input fields and sets the value', () => {
+    fixture.detectChanges();
+    sortCodeFieldInstance.first.setValue('10');
+    sortCodeFieldInstance.second.setValue('20');
+    sortCodeFieldInstance.third.setValue('30');
+    expect(component.form.controls.sortCode.value).toEqual('102030');
+  });
+
+  describe('disabling fields', () => {
+    beforeEach(() => {
+      fixture.detectChanges();
+    });
+
+    it('disables the first input field when the disabled property is set', () => {
+      expect(firstInput.nativeElement.disabled).toEqual(false);
+      component.disabled = true;
+      fixture.detectChanges();
+      expect(firstInput.nativeElement.disabled).toEqual(true);
+    });
+
+    it('disables the second input field when the disabled property is set', () => {
+      expect(secondInput.nativeElement.disabled).toEqual(false);
+      component.disabled = true;
+      fixture.detectChanges();
+      expect(secondInput.nativeElement.disabled).toEqual(true);
+    });
+
+    it('disables the third input field when the disabled property is set', () => {
+      expect(thirdInput.nativeElement.disabled).toEqual(false);
+      component.disabled = true;
+      fixture.detectChanges();
+      expect(thirdInput.nativeElement.disabled).toEqual(true);
+    });
+  });
+
+  it('sets unique identifiers for each input field', () => {
+    fixture.detectChanges();
+    expect(/lg-input-sort-code-first-\d{1,3}/.test(firstInput.nativeElement.id)).toBe(
+      true,
+    );
+    expect(/lg-input-sort-code-second-\d{1,3}/.test(secondInput.nativeElement.id)).toBe(
+      true,
+    );
+    expect(/lg-input-sort-code-third-\d{1,3}/.test(thirdInput.nativeElement.id)).toBe(
+      true,
+    );
+  });
+
+  it('replaces empty fields with an empty string if sort code is not complete', (done) => {
+    fixture.detectChanges();
```

Comment:
I think you could move all these `fixture.detectChanges()` in a before each.

Reply:
Yes, good point 👍

Reply:
Done ✅